pipeline_dp/contribution_bounders.py

```diff
@@ -0,0 +1,145 @@
+import abc
+import collections
+from typing import Callable
+
+import pipeline_dp
+from pipeline_dp import pipeline_backend
+
+
+class ContributionBounder(abc.ABC):
+    """Interface for objects which perform contribution bounding."""
+
+    @abc.abstractmethod
+    def bound_contributions(self, col, params: pipeline_dp.AggregateParams,
+                            backend: pipeline_backend.PipelineBackend,
+                            aggregate_fn: Callable):
+        """Bound contributions of privacy id.
+
+        Contribution bounding is performed to ensure that sensitivity of the
+        query to anonymize is limited. There are many ways how to perform
+        contribution bounding. Sub-classes implement specific contribution
+        strategies.
+
+        Also this function performs the aggregation step per
+        (privacy_id, partition_key) by calling aggregate_fn on data points
+        corresponding to each (privacy_id, partition_key).
+
+        Args:
+          col: collection, with types of each element: (privacy_id,
```

Comment:
For simplicity of the code it has to have. Since `aggregate()` can process any metrics, it's attempted to minimize divergence. In case of count and if the user doesn't specify `value_extractor`, value will be `None`.